src/services/infra/DynamoDBService.ts

```diff
@@ -0,0 +1,72 @@
+import autoBind from "auto-bind"
+
+import { config } from "@config/config"
+
+import {
+  MessageBody,
+  SiteLaunchLambdaStatus,
+} from "@root/../microservices/site-launch/shared/types"
+
+import DynamoDBClient, { UpdateParams } from "./DynamoDBClient"
+
+export default class DynamoDBService {
+  private readonly dynamoDBClient: DynamoDBClient
+
+  private readonly TABLE_NAME: string
+
+  // todo: delete these mock after integration
+  private mockLaunch: MessageBody = {
+    repoName: "my-repo",
+    appId: "my-app",
+    primaryDomainSource: "example.com",
+    primaryDomainTarget: "myapp.example.com",
+    domainValidationSource: "example.com",
+    domainValidationTarget: "myapp.example.com",
+    requestorEmail: "john@example.com",
+    agencyEmail: "agency@example.com",
+    githubRedirectionUrl: "https://github.com/my-repo",
+    redirectionDomain: [
+      {
+        source: "example.com",
+        target: "myapp.example.com",
+        type: "A",
+      },
+    ],
+    status: SiteLaunchLambdaStatus.PENDING,
+  }
+
+  constructor() {
+    this.dynamoDBClient = new DynamoDBClient()
+    this.TABLE_NAME = config.get("aws.dynamodb.siteLaunchTableName")
+    autoBind(this)
+  }
+
+  async createItem(message: MessageBody): Promise<void> {
+    await this.dynamoDBClient.createItem(this.TABLE_NAME, this.mockLaunch)
+  }
+
+  async getItem(message: MessageBody) {
+    return this.dynamoDBClient.getItem(this.TABLE_NAME, this.mockLaunch.appId)
+  }
+
+  async updateItem(message: MessageBody) {
+    // todo: delete mocking after integration
+    this.mockLaunch.status = SiteLaunchLambdaStatus.SUCCESS
+    const updateParams: UpdateParams = {
+      TableName: this.TABLE_NAME,
+      Key: { appId: this.mockLaunch.appId },
+      UpdateExpression: "set #s = :s",
+      ExpressionAttributeNames: { "#s": "status" },
+      ExpressionAttributeValues: {
+        ":s": this.mockLaunch.status,
+      },
```

Comment:
could we have a short comment for what these do? this is cos the `<#|:>s` isn't very understandable

Reply:
ok, added comments for clarity 763b411d